File: api/src/main/java/me/kvdpxne/dtm/statistics/IdentifiableStatistics.java
package me.kvdpxne.dtm.statistics;

import java.io.Serializable;
import me.kvdpxne.dtm.Identifiable;
import org.jetbrains.annotations.NotNull;

public interface IdentifiableStatistics<T extends Serializable>
  extends
  Identifiable<T>,
  Statistics {

  @Override
  @NotNull
  T getIdentifier();
}

File: src/main/kotlin/me/kvdpxne/dtm/statistics/IdentifiableStatistics.kt
package me.kvdpxne.dtm.statistics

import java.util.UUID
import me.kvdpxne.dtm.shared.Identifiable

/**
 * Defines a contract for player statistics that can be measured, compared,
 * and uniquely identified.
 *
 * This interface extends both `Statistics` and `Identifiable`. Statistics
 * objects can provide a measurable value (likely representing an overall
 * summary of the player's performance) and have a unique string identifier
 * for easy retrieval.
 *
 * @since 0.1.0
 */
interface IdentifiableStatistics : Identifiable<UUID>, Statistics